drivers: bluetooth: hci: Use TRNG peripheral for BLE purpose on WB0x

HoZHel opened the pull request:
Call entropy APIs to use TRNG peripheral on STM32WB0x devices for BLE purposes.

Enable `rng` node on Nucleo-WB0x boards.

Supplementary [PR#267](https://github.com/zephyrproject-rtos/hal_stm32/pull/267)

HoZHel committed:
drivers: bluetooth: hci: Use TRNG peripheral for BLE purpose on WB0x

Call entropy APIs to use TRNG peripheral on STM32WB0x devices for BLE
purposes.

Enable RNG node on Nucleo-WB0x boards.

Remove RNG initialization as it's done in the entropy driver.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

HoZHel committed:
west.yml: Update west to point to the recent changes for hal_stm32

Update west.yml to point to the recent changes for hal_stm32.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

mathieuchopstm commented on drivers/bluetooth/hci/hci_stm32wb0.c:
```diff
@@ -301,6 +302,23 @@ static void ble_isr_installer(void)
 }
 
+/* BLEPLAT_RngGetRandomXX definitions are needed for the BLE library. */
+void BLEPLAT_RngGetRandom16(uint16_t *num)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_DRV_INST(0));
+	uint16_t len = 2;
+
+	entropy_get_entropy_isr(dev, (uint8_t *)num, len, 0);
+}
```
Use `ENTROPY_BUSYWAIT` and don't hardcode sizes unnecessarily:
```suggestion
void BLEPLAT_RngGetRandom16(uint16_t *num)
{
	const struct device *dev = DEVICE_DT_GET(DT_DRV_INST(0));

	entropy_get_entropy_isr(dev, (uint8_t *)num, sizeof(*num), ENTROPY_BUSYWAIT);
}
```

---

The following might however provide a lower latency (because it doesn't require turning on the TRNG and polling if entropy is available in the pool):
```suggestion
void BLEPLAT_RngGetRandom16(uint16_t *num)
{
	const struct device *dev = DEVICE_DT_GET(DT_DRV_INST(0));

        /* try to allocate from pool */
	int res = entropy_get_entropy_isr(dev, (uint8_t *)num, sizeof(*num), 0);
	if (res != sizeof(*num)) {
		/* not enough entropy available - fall back to polling */
		entropy_get_entropy_isr(dev, (uint8_t*)num, sizeof(*num), ENTROPY_BUSYWAIT);
	}
}
```

Ditto for BLEPLAT_RngGetRandom32().

This whole sequence could be wrapped in a single function:
```c
static void rng_get_random(void *num, size_t size)
{
	const struct device *dev = DEVICE_DT_GET(DT_DRV_INST(0));

        /* try to allocate from pool */
	int res = entropy_get_entropy_isr(dev, (uint8_t *)num, size, 0);
	if (res != sizeof(*num)) {
		/* not enough entropy available - fall back to polling */
		entropy_get_entropy_isr(dev, (uint8_t*)num, size, ENTROPY_BUSYWAIT);
	}
}
```
and implement the BLEPLAT functions as:
```c
void BLEPLAT_RngGetRandom32(uint32_t *num)
{
    rng_get_random(num, sizeof(*num));
}
```

mathieuchopstm replied:
Right, there was a mistake in `rng_get_random` I suggested, it should compare `res != size` rather than `res != sizeof(*num)`. Edited my suggestion.

erwango commented on west.yml:
```diff
@@ -245,7 +245,7 @@ manifest:
         - hal
     - name: hal_stm32
-      revision: 6e4716f8ac2c6e6159a958a836e1dcca06804456
+      revision: f319c70417cf5d42e25a23f89c1c043c092d4aad
```
This should be `pull/267/head`

HoZHel committed:
soc: stm32: stm32wb0x: Restore main stack size to the default value

Restore main stack size to the default value since fake entropy
implementation from mbedtls is replaced by STM32 entropy driver.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

mathieuchopstm reviewed (commented):
Otherwise LGTM
